src/foam/nanos/dig/DUGRuleAction.js

```diff
@@ -45,68 +49,122 @@ foam.CLASS({
   ],
 
   methods: [
+    {
+      name: 'getLogger',
+      type: 'foam.nanos.logger.Logger',
+      args: [ { name: 'x', type: 'Context' } ],
+      javaCode: `
+        return new PrefixLogger(new Object[] {
+          this.getClass().getSimpleName()
+        }, (Logger) x.get("logger"));
+      `
+    },
     {
       name: 'applyAction',
       javaCode: `
       final var dugRule = (DUGRule) rule;
-      ((Logger) x.get("logger")).debug(this.getClass().getSimpleName(), "Sending DUG webhook", obj);
-
-      agency.submit(x, new ContextAgent() {
-        @Override
-        public void execute(X x) {
-          PM pm = new PM(getClass(), rule.getDaoKey(), rule.getName());
-          DAO dugDigestConfigDAO = (DAO) x.get("dugDigestConfigDAO");
-          DUGDigestConfig dugDigestConfig = (DUGDigestConfig) dugDigestConfigDAO.find(rule.getSpid());
-          try {
-            AbstractSink sink = null;
-            if ( dugDigestConfig != null && dugDigestConfig.getEnabled() ) {
-              sink = new HTTPDigestSink(
-                dugRule.getUrl(),
-                dugRule.evaluateBearerToken(),
-                dugDigestConfig,
-                dugRule.getFormat(),
-                new foam.lib.AndPropertyPredicate(
-                  x,
-                  new foam.lib.PropertyPredicate[] {
-                    new foam.lib.ExternalPropertyPredicate(),
-                    new foam.lib.NetworkPropertyPredicate(),
-                    new foam.lib.PermissionedPropertyPredicate()
-                  }
-                ),
-                true,
-                true
-              );
-            } else {
-              sink = new HTTPSink(
-                dugRule.getUrl(),
-                dugRule.evaluateBearerToken(),
-                dugRule.getFormat(),
-                new foam.lib.AndPropertyPredicate(
-                  x,
-                  new foam.lib.PropertyPredicate[] {
-                    new foam.lib.ExternalPropertyPredicate(),
-                    new foam.lib.NetworkPropertyPredicate(),
-                    new foam.lib.PermissionedPropertyPredicate()
-                  }
-                ),
-                true
-              );
-            }
-
-            sink.setX(x);
-            sink.put(obj, null);
-          } catch (Throwable t) {
-            ((Logger) x.get("logger")).error(this.getClass().getSimpleName(), "Error Sending DUG webhook", t);
-            var alarmDAO = (DAO) x.get("alarmDAO");
-            alarmDAO.put(
-              new Alarm.Builder(x)
-                .setName("DUG/"+rule.getDaoKey() + "/" + rule.getName())
-                .build()
+
+      if ( dugRule.getActAsUser() ) {
```

Comment:
in adding this back, you'll have to set the flag to true manually in the rules.jrl file in the other PR. I also don't see you setting the SecureDaoKey in that PR either.

Reply:
Will do